### ISSUE TITLE
Monthly + byWeekDays + getInstances = fails assertion

### ISSUE THREAD

**Issue by Merrit** [JonasWanke/rrule#62]
1. Create a monthly rrule that specifies `byWeekDays`.
2. Get instances with the start time as `now()`.
3. Any attempt to access the iterable of instances throws an exception due to a failed assertion.

**Expected**

Instances in the future

**Actual**

Failed assertion:

```
_AssertionError ('package:rrule/src/iteration/date_set_filtering.dart': Failed assertion: line 143 pos 10: 'current == date': is not true.)
```

## Example:

```dart
import 'package:rrule/rrule.dart';

void main() {
  final rrule = RecurrenceRule(
    frequency: Frequency.monthly,
    byWeekDays: {ByWeekDayEntry(DateTime.friday, 1)},
  );

  final Iterable<DateTime> instances = rrule.getInstances(
    start: DateTime.now().copyWith(isUtc: true),
  );

  final DateTime nextDueDate = instances.first.copyWith(isUtc: false);
}
```

**Environment:**

- Package version: 0.2.13
- Flutter: 3.13.3
- Dart: 3.1.1

**Comment by JonasWanke** [comment]
This is fixed as part of [v0.2.14](https://pub.dev/packages/rrule/changelog#0214--2023-09-13)

**Comment by Merrit** [comment]
Seems to be working, thank you so much for the fix!

Thank you also for the great package :tada: 💙
